test/tests/executioners/nullspace/gpt_contaminated.i

```diff
@@ -0,0 +1,116 @@
+[Mesh]
+ type = GeneratedMesh
+ dim = 1
+ xmin = 0
+ xmax = 10
+ nx = 8
+[]
+
+[Problem]
+  dimNullSpace = 1
+  dimTransposeNullSpace = 1
+[]
+
+[Variables]
+  [./u]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+[]
+
+[Kernels]
+  [./diff]
+    type = Diffusion
+    variable = u
+  [../]
+
+  [./eig]
+    type = MassEigenKernel
+    variable = u
+    eigen_postprocessor = 1.0002920196258376e+01
+    eigen = false
+  [../]
+
+  [./force]
+    type = CoupledForce
+    variable = u
+    v = aux_v
+  [../]
+[]
+
+[AuxVariables]
+  [./aux_v]
+    order = FIRST
+    family = LAGRANGE
+    [./InitialCondition]
+      type = FunctionIC
+      function = eigen_mode
+    [../]
+  [../]
+[]
+
+[AuxKernels]
+  [./set_source]
+    type = FunctionAux
+    variable = aux_v
+    function = contaminated_second_harmonic
+    execute_on = timestep_begin
+  [../]
+[]
+
+[Functions]
+  [./eigen_mode]
+    type = ParsedFunction
+    value = 'sqrt(2.0 / L) * sin(mode * pi  * x / L)'
+    vars = 'L  mode'
+    vals = '10 1'
+  [../]
+
+  [./contaminated_second_harmonic]
+    type = ParsedFunction
+    value = 'sqrt(2.0 / L) * sin(mode * pi  * x / L) + a * sqrt(2.0 / L) * sin(pi * x / L)'
+    vars = 'L  mode a'
+    vals = '10 2    1'
+  [../]
+[]
+
+[BCs]
+  [./homogeneous]
+    type = DirichletBC
+    variable = u
+    boundary = '0 1'
+    value = 0
+  [../]
+[]
+
+[VectorPostprocessors]
+  [./sample_solution]
+    type = LineValueSampler
+    variable = u
+    start_point = '0 0 0'
+    end_point = '10 0 0'
+    sort_by = x
+    num_points = 9
+    execute_on = timestep_end
+  [../]
+[]
+
+[Preconditioning]
+  [./prec]
+    type = SMP
+    full = true
+  [../]
+[]
+
+[Executioner]
+  type = SteadyWithNull
+  petsc_options_iname = '-pc_type -pc_hypre_type'
+  petsc_options_value = 'hypre boomeramg'
```

Comment:
> it gives an answer that is good to within the printed linear residual. Here's the convergence monitor though that I have been basing my whole argument on all along:
> 
>  0 Nonlinear |R| = 1.484994e+00
>       0 Linear |R| = 2.152831e+00
>       1 Linear |R| = 1.249625e-08
>  1 Nonlinear |R| = 1.089666e+00
> 
> I assume that the 1 Nonlinear is meaningless in this case?

Right, please just disregard `1 Nonlinear |R| = 1.089666e+00` and your problem is essentially a linear problem so that  `1 Linear |R| = 1.249625e-08` makes sense for you.